src/Azure.Functions.ArtifactAssembler/ArtifactAssembler.cs

```diff
@@ -181,11 +207,125 @@ private async Task CreateVisualStudioCoreToolsAsync()
+                string newInProc8ArtifactDirPath = RenameInProcDirectory(inProc8ArtifactDirPath, outOfProcVersion);
+
+                // Copy in-proc8 files and delete old directory
+                await Task.Run(() => FileUtilities.CopyDirectory(newInProc8ArtifactDirPath, Path.Combine(consolidatedArtifactDirPath, InProc8DirectoryName)));
+                Directory.Delete(newInProc8ArtifactDirPath, true);
+
+                // Rename inproc6 directory to have the same version as the out-of-proc artifact before copying
+                var inProcArtifactName = Path.GetFileName(inProc8ArtifactDirPath);
+                var inProc6ArtifactDirPath = Path.Combine(_inProc6ExtractedRootDir, inProcArtifactName);
+                EnsureArtifactDirectoryExist(inProc6ArtifactDirPath);
+                string newInProc6ArtifactDirPath = RenameInProcDirectory(inProc6ArtifactDirPath, outOfProcVersion);
+
+                // Copy in-proc6 files and delete old directory
+                await Task.Run(() => FileUtilities.CopyDirectory(newInProc6ArtifactDirPath, Path.Combine(consolidatedArtifactDirPath, InProc6DirectoryName)));
+                Directory.Delete(newInProc6ArtifactDirPath, true);
+            }
 
             // Delete the extracted directories
             Directory.Delete(_inProc6ExtractedRootDir, true);
             Directory.Delete(_inProc8ExtractedRootDir, true);
             Directory.Delete(_coreToolsHostExtractedRootDir, true);
+            Directory.Delete(_outOfProcExtractedRootDir, true);
+
+            Console.WriteLine($"Finished assembling CLI Core Tools artifacts");
+        }
+
+        private (string artifactDirectory, string version) GetArtifactDirectoryAndVersionNumber(string extractedRootDirectory, string artifactName)
+        {
+            var artifactDirPath = Directory.EnumerateDirectories(extractedRootDirectory)
+                                          .FirstOrDefault(dir => dir.Contains(artifactName));
+            if (artifactDirPath == null)
+            {
+                throw new InvalidOperationException($"Artifact directory '{artifactDirPath}' not found!");
+            }
+
+            var version = GetCoreToolsProductVersion(artifactDirPath);
+            return (artifactDirPath, version);
+        }
+
+        private string RenameInProcDirectory(string oldArtifactDirPath, string newVersion)
+        {
+            string pattern = @"^(.*?)(\d+\.\d+\.\d+)$";  // Capture everything before the version
+
+            Match match = Regex.Match(oldArtifactDirPath, pattern);
+
+            if (!match.Success)
+            {
+                throw new InvalidOperationException($"Unable to extract content before version number from '{oldArtifactDirPath}'.");
+            }
+
+            var contentBeforeVersion = match.Groups[1];
+            var newDirectoryName =  $"{contentBeforeVersion}{newVersion}";
```

Comment:
is content before version just the artifact name? can we rename the variable to something more descriptive?

Reply:
renamed!